herbieapp/services/message_publisher/google_pub_sub_publisher.py

```diff
@@ -0,0 +1,57 @@
+import logging
+from google.api_core.exceptions import AlreadyExists
+from google.cloud.pubsub_v1 import PublisherClient, SubscriberClient
+from herbie import settings
+from rest_framework.renderers import JSONRenderer
+
+from herbieapp.models.message_models_serializers import EntityUpdateMessage, EntityUpdateMessageSerializer, \
+    EntityDeleteMessage, EntityDeleteMessageMessageSerializer
+
+
+class GooglePubSubPublisher:
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._logger = logging.getLogger(__name__)
+        self._publisher = PublisherClient()
+        self._subscriber = SubscriberClient()
+
+    def send_message(self, message):
+        serializer = self._get_serializer(message)
+
+        json_data = JSONRenderer().render(serializer.data)
+
+        topic_path = self._publisher.topic_path(settings.GCLOUD_PUBSUB_PROJECT_ID, 'customer')
+        future = self._publisher.publish(topic_path, data=json_data)
+        future.add_done_callback(self._publish_callback)
+
+    def _publish_callback(self, message_future):
+        if message_future.exception(timeout=30):
+            self._logger.info(f'Publishing message on {message_future.exception()} threw an Exception.')
+        else:
+            self._logger.info(message_future.result())
+
+    def create_topic(self, topic: str):
+        topic_path = self._publisher.topic_path(settings.GCLOUD_PUBSUB_PROJECT_ID, topic)
+
+        try:
+            self._publisher.create_topic(topic_path)
+        except AlreadyExists as e:
+            self._logger.info(f'Topic {topic} already exists.')
+
+    def create_subscription(self, topic: str, subscription: str):
+        topic_path = self._subscriber.topic_path(settings.GCLOUD_PUBSUB_PROJECT_ID, topic)
+        subscription_path = self._subscriber.subscription_path(settings.GCLOUD_PUBSUB_PROJECT_ID, subscription)
+
+        try:
+            self._subscriber.create_subscription(subscription_path, topic_path)
+        except AlreadyExists as e:
+            self._logger.info(f'Subscription {subscription} already exists.')
+            pass
+
+    def _get_serializer(self, message):
```

Comment:
Just a small idea: you could add a `get_serializer` method to the message classes, then you don't need this method here.
But I'm also not sure why you even need those Serializers.

Reply:
At first we need the serializers in order to bring a python object to python primitive types (dictionary). The reason for that is because google pub sub needs to bring the data in json format before send the message as you can see in send_message function of GooglePubSubPublisher.
We could add a get_serializer function in the classes of message_types (EntityUpdateMessage, EntityDeleteMessage) but i really prefer classes that keep data to be clean and not have any extra functionality. Also this is the only place that we need to retrieve the serializer of the objects. In the future if we need again the same functionality(retrieving dynamically the serializer) we can move this function to util class. What do you think? Is something that i miss?